### ISSUE TITLE
Feature Request: nth Parent Selector

### ISSUE THREAD

**Issue by x6eull** [sass/sass#3637]
I want to use "the parent of the current parent" in a nested selector.
A use case looks like:
```scss
#root{
    .todo-item{
        .mark{
            // some rules
        
            // I want to select "#root .todo-item:hover .mark"
            &&:hover &{
                background: gray;
            }
        }
    }
}
```
I have found no approach to make this into reality.
For now, the only way to select "#root .todo-item:hover .mark" is using `&:hover .mark` in `.todo-item`'s rules. It's a bit wordy, while a practical nth Parent Selector would make the code clearer and more reasonable as `background: gray` should belong to `.mark`.

I've read #2690, #2264 and [sass:selector module document](https://sass-lang.com/documentation/modules/selector). I ran into `$selector: Parent selectors aren't allowed here.` error when I tried to find a way out. It seems that there's no perfect workaround to achieve this (@at-root would discard `#root` in the case). It there is, please mention it. If it could be achieved using selector functions or something alike, consider adding a implement as a build-in function. It will be pretty useful!

Thanks for the time. Apologies for any mistakes as English isn't my native language.

**Comment by nex3** [comment]
Sass takes pains to make sure there's no difference between writing `.foo { .bar { ... } }` and `.foo .bar { ... }`. This would break that encapsulation, and potentially open the door to confusing behavioral changes for transformations authors (reasonably) expect to be no-ops.

If you want to get the parent selector _without the last descendant selector_, that's doable with `sass:selector`, [as in this example](https://sass-lang.com/playground/#MTElNDB1c2UlMjAnc2FzcyUzQXNlbGVjdG9yJyUzQiUwQSU0MHVzZSUyMCdzYXNzJTNBbGlzdCclM0IlMEElMEElNDBmdW5jdGlvbiUyMHBhcmVudC13aXRob3V0LWxhc3QoKSUyMCU3QiUwQSUyMCUyMCU0MGlmJTIwbGlzdC5sZW5ndGgoJTI2KSUyMCUzRSUyMDElMjAlN0IlMEElMjAlMjAlMjAlMjAlNDBlcnJvciUyMCUyMkNhbid0JTIwY2FsbCUyMHBhcmVudC13aXRob3V0LWxhc3QoKSUyMG9uJTIwYSUyMHNlbGVjdG9yJTIwbGlzdCUyMHdpdGglMjAlMjIlMjAlMkIlMEElMjAlMjAlMjAlMjAlMjAlMjAlMjJtb3JlJTIwdGhhbiUyMG9uZSUyMHNlbGVjdG9yLiUyMiUzQiUwQSUyMCUyMCU3RCUwQSUwQSUyMCUyMCUyNHNlbGVjdG9yJTNBJTIwbGlzdC5udGgoJTI2JTJDJTIwMSklM0IlMEElMjAlMjAlMjRyZXN1bHQlM0ElMjAoKSUzQiUwQSUyMCUyMCU0MGZvciUyMCUyNGklMjBmcm9tJTIwMSUyMHRvJTIwbGlzdC5sZW5ndGgoJTI0c2VsZWN0b3IpJTIwJTdCJTBBJTIwJTIwJTIwJTIwJTI0cmVzdWx0JTNBJTIwbGlzdC5hcHBlbmQoJTI0cmVzdWx0JTJDJTIwbGlzdC5udGgoJTI0c2VsZWN0b3IlMkMlMjAlMjRpKSUyQyUyMHNwYWNlKSUzQiUwQSUyMCUyMCU3RCUwQSUyMCUyMCU0MHJldHVybiUyMCUyNHJlc3VsdCUzQiUwQSU3RCUwQSUwQSU0MGZ1bmN0aW9uJTIwbGFzdC1wYXJlbnQoKSUyMCU3QiUwQSUyMCUyMCU0MGlmJTIwbGlzdC5sZW5ndGgoJTI2KSUyMCUzRSUyMDElMjAlN0IlMEElMjAlMjAlMjAlMjAlNDBlcnJvciUyMCUyMkNhbid0JTIwY2FsbCUyMGxhc3QtcGFyZW50KCklMjBvbiUyMGElMjBzZWxlY3RvciUyMGxpc3QlMjB3aXRoJTIwJTIyJTIwJTJCJTBBJTIwJTIwJTIwJTIwJTIwJTIwJTIybW9yZSUyMHRoYW4lMjBvbmUlMjBzZWxlY3Rvci4lMjIlM0IlMEElMjAlMjAlN0QlMEElMEElMjAlMjAlMjRzZWxlY3RvciUzQSUyMGxpc3QubnRoKCUyNiUyQyUyMDEpJTNCJTBBJTIwJTIwJTQwcmV0dXJuJTIwbGlzdC5udGgoJTI0c2VsZWN0b3IlMkMlMjBsaXN0Lmxlbmd0aCglMjRzZWxlY3RvcikpJTNCJTBBJTdEJTBBJTBBJTIzcm9vdCUyMCU3QiUwQSUyMCUyMC50b2RvLWl0ZW0lMjAlN0IlMEElMjAlMjAlMjAlMjAubWFyayUyMCU3QiUwQSUyMCUyMCUyMCUyMCUyMCUyMCUyRiUyRiUyMHNvbWUlMjBydWxlcyUwQSUyMCUyMCUwQSUyMCUyMCUyMCUyMCUyMCUyMCUyRiUyRiUyMEklMjB3YW50JTIwdG8lMjBzZWxlY3QlMjAlMjIlMjNyb290JTIwLnRvZG8taXRlbSUzQWhvdmVyJTIwLm1hcmslMjIlMEElMjAlMjAlMjAlMjAlMjAlMjAlNDBhdC1yb290JTIwJTIzJTdCcGFyZW50LXdpdGhvdXQtbGFzdCgpJTdEJTNBaG92ZXIlMjAlMjMlN0JsYXN0LXBhcmVudCgpJTdEJTIwJTdCJTBBJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwYmFja2dyb3VuZCUzQSUyMGdyYXklM0IlMEElMjAlMjAlMjAlMjAlMjAlMjAlN0QlMEElMjAlMjAlMjAlMjAlN0QlMEElMjAlMjAlN0QlMEElN0Q=). But be aware that it _will_ have the same behavior whether you write your style rules nested or not.
